torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -96,6 +107,33 @@ at::Tensor inferAndAlloc(
 
 } // namespace
 
+uint64_t FusionExecutor::computeSharedMemory(
+    EvaluationContext& ec,
+    const std::vector<kir::Allocate*>& buffers,
+    bool align_padding,
+    uint64_t total) {
+  for (auto smem_alloc : buffers) {
+    auto inferred_size = ExpressionEvaluator::evaluate(smem_alloc->size(), &ec);
+    if (inferred_size.has_value()) {
+      unsigned data_size = dataTypeSize(smem_alloc->buffer_type());
+      // Add padding to align dynamic shared memory
+      if (align_padding && total > 0) {
+        auto aligned = std::ceil(total / data_size) * data_size;
+        total += (aligned - total);
```

Comment:
1. wouldn't `total = ceilDiv(total / data_size) * data_size;` suffice? (btw, using floating point to calculate integer arithmetic can be tricky, at least since the default rounding modes are different)

2. how about we round `total`, after adding the allocation size? This way total is always "aligned" - an easier to check and understand invariant.

Reply:
We might have a shared buffer for reduction/broadcast before the dynamic smem buffers, so we need to round first then add the next size.

Reply:
It matches the flow in the kernel to keep things simple.